frontend/pages/shelter/pet/[id].tsx

```diff
@@ -0,0 +1,345 @@
+		score: 20,
+		status: "pending"
+	},
+	{
+		key: "2",
+		name: "Adopter 2",
+		applicationDate: new Date(2021, 8, 19),
+		score: 17,
+		status: "rejected"
+	},
+	{
+		key: "3",
+		name: "Adopter 3",
+		applicationDate: new Date(2021, 8, 20),
+		score: 20,
+		status: "pending"
+	},
+	{
+		key: "4",
+		name: "Adopter 4",
+		applicationDate: new Date(2021, 8, 21),
+		score: 45,
+		status: "pending"
+	},
+	{
+		key: "5",
+		name: "Adopter 5",
+		applicationDate: new Date(2021, 8, 22),
+		score: 28,
+		status: "pending"
+	},
+	{
+		key: "6",
+		name: "Adopter 6",
+		applicationDate: new Date(2021, 8, 25),
+		score: 62,
+		status: "pending"
+	}
+];
+
+const adopterCols = [
+	{
+		title: "Application Id",
+		dataIndex: "key",
+		key: "key",
+		render: (text: Adopter["key"]) => <span>{text}</span>,
+		sorter: (a: Adopter, b: Adopter) => a.key.localeCompare(b.key)
+	},
+	{
+		title: "Name",
+		dataIndex: "name",
+		key: "name",
+		render: (text: Adopter["name"]) => (
+			<div style={{ display: "flex", alignItems: "center" }}>
+				<Image
+					objectFit="cover"
+					src="https://via.placeholder.com/22"
+					alt="Logo"
+					width="22"
+					height="22"
+				/>
+				<span style={{ marginLeft: 8 }}>{text}</span>
+			</div>
+		),
+		sorter: (a: Adopter, b: Adopter) => a.name.localeCompare(b.name)
+	},
+	{
+		title: "Date Of Application",
+		dataIndex: "applicationDate",
+		key: "applicationDate",
+		render: (applicationDate: Adopter["applicationDate"]) => (
+			<span>{dayjs(applicationDate).format("DD/MM/YYYY")}</span>
+		),
+		sorter: (a: Adopter, b: Adopter) =>
+			a.applicationDate.getTime() - b.applicationDate.getTime()
+	},
+	{
+		title: "Quiz Score",
+		dataIndex: "score",
+		key: "score",
+		render: (text: Adopter["score"]) => <span>{text}</span>,
+		sorter: (a: Adopter, b: Adopter) => a.score - b.score,
+		filters: [
+			{
+				text: ">= 20",
+				value: 20
+			},
+			{
+				text: ">= 40",
+				value: 40
+			}
+		],
+		onFilter: (value: any, record: Adopter) => record.score >= value
+	},
+	{
+		title: "Status",
+		dataIndex: "status",
+		key: "status",
+		render: (status: Adopter["status"]) => {
+			switch (status) {
+				case "rejected":
+					return <Tag color="red">rejected</Tag>;
+				case "pending":
+					return <Tag color="orange">pending</Tag>;
+				default:
+					return <Tag color="purple">unknown</Tag>;
+			}
+		},
+		filters: [
+			// TODO: auto fill this based off an array
+			{
+				text: "Rejected",
+				value: "rejected"
+			},
+			{
+				text: "Pending",
+				value: "pending"
+			}
+		],
+		onFilter: (value: any, record: Adopter) =>
+			record.status.indexOf(value) === 0
+	},
+	{
+		title: "Action",
+		key: "action",
+		render: (_: any, record: Adopter) => (
+			<Space size="middle">
+				<Link href={`adopter/${record.key}`}>
+					<a>View Details</a>
+				</Link>
+			</Space>
+		)
+	}
+];
+
+export default function PetDetails() {
+	const router = useRouter();
+	const { id } = router.query;
```

Comment:
Will the `id` be correct since "Route: /shelter/pet/:id" seems to be using the path instead of query param>

Reply:
yes, this is according to nextJS docs
[https://nextjs.org/docs/routing/dynamic-routes](https://nextjs.org/docs/routing/dynamic-routes)